OpenSim/Common/Component.cpp

```diff
@@ -134,11 +152,19 @@ void Component::finalizeFromProperties()
 }
 
 // Base class implementation of virtual method.
-// Call extendFinalizeFromProperties on all components
+// Call finalizeFromProperties on all subcomponents
 void Component::componentsFinalizeFromProperties() const
```

Comment:
Is it possible (through some circuitous route) for this method to result in an infinite loop of `finalizeFromProperties()` calls? For example, is there anything preventing an adopted subcomponent from adopting its grandparent? I would have expected something like
```cpp
if (!_finalizedFromProperties) {
    _finalizedFromProperties = true;
    ...
}
```
I'm still learning the Component API, so please ignore if the machinery is already in place to prevent this. :frog: